drivers/storage/portworx/component/telemetry.go

```diff
@@ -886,6 +913,63 @@ func (t *telemetry) createDaemonSetTelemetryPhonehome(
 	return nil
 }
 
+func (t *telemetry) createDeploymentTelemetryCollectorV2(
+	cluster *corev1.StorageCluster,
+	ownerRef *metav1.OwnerReference,
+) error {
+	deployment, err := k8sutil.GetDeploymentFromFile(deploymentFileNameTelemetryCollectorV2, pxutil.SpecsBaseDir())
```

Comment:
one easy way is to save this deployment so we dont need to reload it from disk every time.

Reply:
I have a ticket open for this, need to do this enhancement for all configmaps etc. I'm thinking read those into objects only upon operator start once.